From 6201eb57e6f556b4cca5d6775c3d5e3a3b4ad1e1 Mon Sep 17 00:00:00 2001
From: Shraddha Basantwani <basantwani@google.com>
Date: Mon, 2 Dec 2024 09:36:26 +0000
Subject: [PATCH] Add a list of Biometric protected packages

Protected packages require biometric authentication for modification
(Disable, force-stop or uninstalling updates).

Bug: 352504490, 344865740
Test: atest AppButtonsPreferenceControllerTest PackageInfoPresenterTest
Flag: EXEMPT High Security Bug
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:10e65babc1ca2101508c251e3e39de4cda62f011)
(cherry picked from https://googleplex-android-review.googlesource.com/q/commit:97c0dc4f12574c6e9cb6422af12a4706712664eb)
Merged-In: I0c494e307b02229d751de118abcc89e4e61a6861
Change-Id: I0c494e307b02229d751de118abcc89e4e61a6861
---
 core/res/res/values/config.xml  | 4 ++++
 core/res/res/values/symbols.xml | 3 +++
 2 files changed, 7 insertions(+)

diff --git a/core/res/res/values/config.xml b/core/res/res/values/config.xml
index 7bdee77ebbf2..1a9a4178e4ec 100644
--- a/core/res/res/values/config.xml
+++ b/core/res/res/values/config.xml
@@ -7125,6 +7125,10 @@
          OEM/Partner can explicitly opt to disable the screen. -->
     <bool name="config_enableGaiaEducationInPrivateSpace">true</bool>
 
+    <!-- List of protected packages that require biometric authentication for modification
+         (Disable, force-stop or uninstalling updates). -->
+    <string-array name="config_biometric_protected_package_names" translatable="false" />
+    
     <!-- Whether to enable usb state update via udc sysfs. -->
     <bool name="config_enableUdcSysfsUsbStateUpdate">false</bool>
 
diff --git a/core/res/res/values/symbols.xml b/core/res/res/values/symbols.xml
index 06b36b8f74af..42e70352cd67 100644
--- a/core/res/res/values/symbols.xml
+++ b/core/res/res/values/symbols.xml
@@ -5590,6 +5590,9 @@
   <java-symbol type="drawable" name="ic_zen_mode_type_unknown" />
   <java-symbol type="drawable" name="ic_zen_mode_type_special_dnd" />
 
+  <!-- List of protected packages that require biometric authentication for modification -->
+  <java-symbol type="array" name="config_biometric_protected_package_names" />
+
   <!-- System notification for background user sound -->
   <java-symbol type="string" name="bg_user_sound_notification_title_alarm" />
   <java-symbol type="string" name="bg_user_sound_notification_button_switch_user" />
-- 
2.34.1

